templates/forms/license-activation.php

```diff
@@ -59,13 +59,60 @@
     if ( $is_network_activation ) {
         $all_sites = Freemius::get_sites();
 
+        $install_count_by_id_map     = array();
+        $site_info_by_install_id_map = array();
+
         foreach ( $all_sites as $site ) {
-            $site_details = $fs->get_site_info( $site );
+            $blog_id = Freemius::get_site_blog_id( $site );
+            $install = $fs->get_install_by_blog_id( $blog_id );
+
+            if ( is_object( $install ) ) {
+                /**
+                 * The install counter is used in determining if an install has a duplicate.
+                 *
+                 * @author Leo Fajardo (@leorw)
+                 * @since 2.5.0
+                 */
+                $install_count_by_id_map[ $install->id ] = isset( $install_count_by_id_map[ $install->id ] ) ?
+                    ( $install_count_by_id_map[ $install->id ] = $install_count_by_id_map[ $install->id ] + 1 ) :
+                    1;
+
+                if ( $install_count_by_id_map[ $install->id ] > 1 ) {
+                    continue;
+                }
+            }
 
+            $site_info_by_install_id_map[] = array(
+                'site'    => $site,
+                'install' => $install,
+            );
+        }
+
+        foreach ( $site_info_by_install_id_map as $site_info )

```

Comment:
@fajardoleo similar to one of the previous comments in `get_installs_data_for_api()`, what's the purpose of the additional `foreach`? why not handle the logic within the first `foreach`?

Reply:
@vovafeldman The main purpose is to prevent removing entries from the `$sites_details` array if later on, we find a subsite with a clone install.